src/oc/auth/api/users.clj

```diff
@@ -53,17 +53,32 @@
           (timbre/info "Failed to auth:" email) 
           false)))))
 
-(defn- allow-user-and-team-admins [conn {accessing-user-id :user-id} accessed-user-id]
-  (or
-    ;; JWToken user-id matches URL user-id, user accessing themself
-    (= accessing-user-id accessed-user-id)
-
-    ;; check if the accessing user is an admin of any of the accessed user's teams
-    (if-let* [accessed-user (user-res/get-user conn accessed-user-id)
-              teams (team-res/list-teams-by-ids conn (:teams accessed-user) [:admins])]
-      (some #((set (:admins %)) accessing-user-id) teams)
+(defn- allow-superuser-token
+  [ctx]
+  (let [token (api-common/get-token (get-in ctx [:request :headers]))]
+    (if-let [decoded-token (jwt/decode token)]
+      (do
+        (if (and (jwt/check-token token config/passphrase)    ;; We signed the token
+                 (:super-user (:claims decoded-token)))
+          {:jwtoken decoded-token :user (:claims decoded-token)}
+          false))
       false)))
```

Comment:
This code doesn't use `oc.lib.api.common/open-company-authenticated-resource`.  `:initialize-context` is where most of that is done but in this case it is overridden and the code here is used.   More reason to put this code into `oc.lib`

Reply:
I see because you override `:initialize-context` at line 255. I see you have the JWT set in the same place of a normal request, why you don't add a check for superusers in `oc.lib.api.common`, like `open-company-authenticated-resource` does but it also checks if it's a superuser token.

It will be useful on other services probably.

I'll leave this review to @belucid now, just a suggestion.

Reply:
maybe make a tech debt card to move the super user token into lib?